From 65acc8800dba7e10da882871d4648241805c47ce Mon Sep 17 00:00:00 2001
From: Andre McCurdy <amccurdy@gmail.com>
Date: Sat, 30 Apr 2016 15:29:06 -0700
Subject: [PATCH] use <asm/sgidefs.h>

Build fix for MIPS with musl libc

The MIPS specific header <sgidefs.h> is provided by glibc and uclibc
but not by musl. Regardless of the libc, the kernel headers provide
<asm/sgidefs.h> which provides the same definitions, so use that
instead.

Upstream-Status: Pending

[Vincent:
Taken from: https://sourceware.org/bugzilla/show_bug.cgi?id=21070]

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>
---
 gdb/mips-linux-nat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mfbt/RandomNum.cpp
+++ b/mfbt/RandomNum.cpp
@@ -52,7 +52,7 @@ extern "C" BOOLEAN NTAPI RtlGenRandom(PV
 #  elif defined(__s390__)
 #    define GETRANDOM_NR 349
 #  elif defined(__mips__)
-#    include <sgidefs.h>
+#    include <asm/sgidefs.h>
 #    if _MIPS_SIM == _MIPS_SIM_ABI32
 #      define GETRANDOM_NR 4353
 #    elif _MIPS_SIM == _MIPS_SIM_ABI64
